qualtran/serialization/sympy_test.py

```diff
@@ -0,0 +1,54 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+import sympy
+
+from qualtran.serialization.bloq import arg_from_proto, arg_to_proto
+
+
+@pytest.mark.parametrize(
+    'expr',
+    [
+        (
+            sympy.parse_expr("5")
+            + sympy.symbols("x")
+            + sympy.parse_expr("1/2")
+            + sympy.pi
+            + sympy.parse_expr("2j")
+        ),
+        (sympy.parse_expr("(-b + sqrt(-4*a*c + b**2))/(2*a)")),
+    ],
+)
```

Comment:
Let's not depend upon `parse_expr` for tests. Instead, let's create a list of expressions; potentially generated by small functions; which we'd like to test and parameterize over it. For example:

```suggestion
x, N = sympy.Symbol('x', positive=True),  sympy.Symbol('N')

# These should return a `sympy_pb2.Parameter` proto object?
sympy_parameters_to_test = [ 
    # Only symbols
    sympy.Symbol('x'),
    sympy.Symbol('N'),
    sympy.Symbol('E'),
    
    # Sympy constants
    sympy.pi, sympy.Infinity, sympy.E, sympy.I # etc.
    
    # Integers, Floats, Rationals
    sympy.Integer(5),
    sympy.Float(0.1),
    sympy.Rational("1/2"),
    sympy.Rational('1/10'),
]
sympy_expr_to_test = [
   5 * x + sympy.sqrt(N),
   # Add different types of functions that are supported, maybe in a single expression or multiple expressions. 
]
@pytest.mark.parametrize('expr', sympy_parameters_to_test + sympy_exprs_to_test)
```